packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts

```diff
@@ -359,11 +373,7 @@ export class EmrCreateCluster extends sfn.TaskStateBase {
   private createServiceRole(): iam.IRole {
     if (cdk.FeatureFlags.of(this).isEnabled(ENABLE_EMR_SERVICE_POLICY_V2)) {
       return new iam.Role(this, 'ServiceRole', {
-        assumedBy: new iam.ServicePrincipal('elasticmapreduce.amazonaws.com', {
-          conditions: {
-            StringEquals: { 'aws:RequestTag/for-use-with-amazon-emr-managed-policies': 'true' },
-          },
-        }),
```

Comment:
This condition isn't needed and was actually causing it to fail.

Reply:
Could you explain your reasoning for why it's not needed? Also how was it causing the integ tests to fail?

Reply:
When included, I get the following error:

```
EMR service role arn:aws:iam::<REDACTED>:role/aws-cdk-emr-create-cluste-EmrCreateClusterServiceRo-GrrjLUAR1a2Q is invalid
```